April Semrush SEO Report (#1127)
moonbeam-foundation/moonbeam-docs: builders/substrate/libraries/py-substrate-interface.md

### Accessing Runtime Constants {: #accessing-runtime-constants }

All runtime constants, such as `BlockWeights`, `DefaultBlocksPerRound` and `ExistentialDeposit`, are provided in the metadata. You can use the [`get_metadata_constants`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/#substrateinterface.base.SubstrateInterface.get_metadata_constants){target=\_blank} method to see a list of available runtime constants within Moonbeam network's metadata.

> Runtime constants available in the metadata can be queried through the [`get_constant`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/#substrateinterface.base.SubstrateInterface.get_constant
> ){target=\_blank} method.

Reviewer: ```suggestion
Runtime constants available in the metadata can be queried through the [`get_constant`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/#substrateinterface.base.SubstrateInterface.get_constant){target=\_blank} method.
```

### Querying for Storage Information {: #querying-for-storage-information }

> You can use the [`get_metadata_storage_functions`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/?h=get+constant+metadata#substrateinterface.base.SubstrateInterface.get_metadata_storage_function){target=\_blank} to see a list of available storage functions within Moonbeam network's metadata.

Reviewer: ```suggestion
You can use the [`get_metadata_storage_functions`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/?h=get+constant+metadata#substrateinterface.base.SubstrateInterface.get_metadata_storage_functions){target=\_blank} to see a list of available storage functions within Moonbeam network's metadata.
```

### Forming and Sending a Transaction {: #forming-and-sending-a-transaction }

> The [`compose_call`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/?h=compose_call#substrateinterface.base.SubstrateInterface.compose_call){target=\_blank} method can be used to compose a call payload which can be used as an unsigned extrinsic or a proposal.

Reviewer: remove the query string

> Then the payload can be signed using a keypair through the [`create_signed_extrinsic`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/?h=create_signed_extrinsic#substrateinterface.base.SubstrateInterface.create_signed_extrinsic){target=\_blank} method.

Reviewer: Remove the query string part of this starting wiht the `?h=`

> The signed extrinsic can then be submitted using the [`submit_extrinsic`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/?h=submi#substrateinterface.base.SubstrateInterface.submit_extrinsic){target=\_blank} method.

Reviewer: ```suggestion
The signed extrinsic can then be submitted using the [`submit_extrinsic`](https://jamdottech.github.io/py-polkadot-sdk/reference/base/#substrateinterface.base.SubstrateInterface.submit_extrinsic){target=\_blank} method.
```

Reviewer: This just removes the query string from the URL

### Offline Signing {: #offline-signing }

> You can sign transaction payloads or any arbitrary data using a keypair object through the [`sign`](https://jamdottech.github.io/py-polkadot-sdk/usage/keypair-creation-and-signing/#keypair-creation-and-signing){target=\_blank} method. This can be used for offline signing of transactions.

Reviewer: ```suggestion
You can sign transaction payloads or any arbitrary data using a keypair object through the [`sign`](https://jamdottech.github.io/py-polkadot-sdk/reference/keypair/#substrateinterface.keypair.Keypair.sign){target=\_blank} method. This can be used for offline signing of transactions.
```
